analysis/generate_corp_decomposition.R

```diff
@@ -0,0 +1,229 @@
+library(ggrepel)
+library(geomtextpath)
+library(glue)
+library(arrow)
+
+# Load the necessary data from the targets store
+tar_load(clean_variant_data_final_all_states)
+all_model_outputs_for_heatmap <- read_parquet(
+    "~/variant-25a.parquet"
+)
+
+# Define function to run analysis and plot
+run_corp_analysis <- function(
+    location_subset_name,
+    loc_filter_fn,
+    model_selector,
+    output_file
+) {
+    cat(glue::glue("Processing {location_subset_name}...\n"))
+
+    # 1. Prepare Observations
+    obs <- clean_variant_data_final_all_states |>
+        filter(loc_filter_fn(location)) |>
+        rename(target_date = date, clade = clades_modeled) |>
+        group_by(location, target_date) |>
+        mutate(total_sequences = sum(sequences)) |>
+        ungroup() |>
+        select(location, target_date, clade, sequences, total_sequences)
+
+    # 2. Prepare Predictions
+    preds_long <- all_model_outputs_for_heatmap |>
+        filter(
+            loc_filter_fn(location),
+            output_type == "mean",
+            clade == "25A"
+        ) |>
+        select(model_id, location, nowcast_date, target_date, clade, value) |>
+        rename(prediction = value)
+
+    # Pivot to wide to handle model selection
+    preds_wide <- preds_long |>
+        pivot_wider(names_from = model_id, values_from = prediction)
+
+    # Apply model selection logic
+    preds_wide <- model_selector(preds_wide)
+
+    # 3. Join Predictions and Observations
+    data_joined_full <- preds_wide |>
+        inner_join(obs, by = c("location", "target_date", "clade"))
+
+    # Identify model columns
+    non_model_cols <- c(
+        "location",
+        "target_date",
+        "clade",
+        "sequences",
+        "total_sequences",
+        "nowcast_date"
+    )
+    model_cols <- setdiff(names(data_joined_full), non_model_cols)
+
+    # Drop rows with NAs in any of the selected models to ensure fair comparison
+    data_joined <- data_joined_full |>
+        drop_na(all_of(model_cols))
+
+    dropped_dates <- setdiff(
+        data_joined_full$nowcast_date,
+        data_joined$nowcast_date
+    )
+    if (length(dropped_dates) > 0) {
+        cli::cli_alert_info(
+            "Dropped nowcast dates {dropped_dates} due to missing model predictions."
+        )
+    }
+
+    if (nrow(data_joined) == 0) {
+        message(glue::glue(
+            "No data available for {location_subset_name} after filtering."
+        ))
+        return(NULL)
+    }
+
+    # 4. Expand data to binary observations
+    # Successes
+    df_success <- data_joined |>
+        select(all_of(c(model_cols, "sequences"))) |>
+        uncount(sequences) |>
+        mutate(y = 1)
+
+    # Failures
+    df_failure <- data_joined |>
+        mutate(failures = total_sequences - sequences) |>
+        select(all_of(c(model_cols, "failures"))) |>
+        uncount(failures) |>
+        mutate(y = 0)
+
+    # Combine
+    df_expanded <- bind_rows(df_success, df_failure)
```

Comment:
Is this required just for evaluating the Brier score since it expects binary data (unless you use the calculation we used in the global variant sequencing paper where you can compute the brier score with the sequence counts by clade and total sequences)?